### ISSUE TITLE
lint-fmt: cannot find a solution for ocamlformat itself

### ISSUE THREAD

**Issue by emillon** [ocurrent/ocaml-ci#261]
Hi,

Context:
The next version of ocamlformat will use ppxlib >= 0.18.0. In the PR where I'm adding this dependency, the lint-fmt job fails:

https://ci.ocamllabs.io/github/ocaml-ppx/ocamlformat/commit/8b1ed80e8b77be4b4c4ba52daf2d8447c3fc6602/variant/(lint-fmt)

```
# Detecting depexts using vars: arch=x86_64, os=linux, os-distribution=debian, os-family=debian
[ERROR] No solution for ocamlformat: The following dependencies couldn't be met:
          - ocamlformat -> ppxlib >= 0.18.0
              no matching version
```

I think there's no matching version because all released versions conflict with that ppxlib version (all released versions depend on omp 1.x, this ppxlib version on omp 2.x).

But I thought that there was special handling so that ocamlformat could use its dev version to format itself, so I'm not sure why it can't find a solution.

Thanks!

**Comment by talex5** [comment]
The problem here is that the lint job (unlike the main build jobs) doesn't do a solve and select a suitable version of opam-repository. Instead, it just tries to install ocamlformat from the base image's snapshot of opam-repository, which at the moment doesn't include ppxlib 0.18.0.

**Comment by emillon** [comment]
FYI, we're planning to release ocamlformat 0.15.1 and 0.16.0 next week which depend on ppxlib 0.18.0 so this is likely to affect more builds.

**Comment by talex5** [comment]
I've updated the base image for the ocamlformat jobs so it now has the necessary version of ppxlib. That should fix the problem for now, although I can't test it at the moment as ocaml-ci has been disabled on that repository (@avsm: could you re-enable this?).

**Comment by talex5** [comment]
Should be fixed by #357.
